### PR TITLE
Update to Dagger2 for sample

### PR DESCRIPTION
**Pull request by michaldrabik**
Dagger version update from 1.2.2 to 2.2 in sample

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -20,9 +20,11 @@ ext {
...
   supportTestRunner = 'com.android.support.test:runner:0.3'
 
-  // Square dependencies.
-  dagger = 'com.squareup.dagger:dagger:1.2.2'
-  daggerCompiler = 'com.squareup.dagger:dagger-compiler:1.2.2'
+  // Dagger dependencies.
+  dagger = 'com.google.dagger:dagger:2.2'
+  daggerCompiler = 'com.google.dagger:dagger-compiler:2.2'
+  javaxAnnotations = 'javax.annotation:jsr250-api:1.0'
+  javaxInject = 'javax.inject:javax.inject:1'
```
Comment (JakeWharton) [inline build.gradle]: This isn't needed. It comes transitively from the `dagger` artifact.

File: build.gradle [hunk 2]
```diff
@@ -20,9 +20,11 @@ ext {
   supportAnnotations = 'com.android.support:support-annotations:23.0.0'
   supportTestRunner = 'com.android.support.test:runner:0.3'
 
-  // Square dependencies.
-  dagger = 'com.squareup.dagger:dagger:1.2.2'
-  daggerCompiler = 'com.squareup.dagger:dagger-compiler:1.2.2'
+  // Dagger dependencies.
+  dagger = 'com.google.dagger:dagger:2.2'
+  daggerCompiler = 'com.google.dagger:dagger-compiler:2.2'
```
Comment (JakeWharton) [inline build.gradle]: 2.4 is latest

File: build.gradle [hunk 3]
```diff
@@ -20,9 +20,11 @@ ext {
...
+  javaxAnnotations = 'javax.annotation:jsr250-api:1.0'
```
Comment (JakeWharton) [inline build.gradle]: This shouldn't be needed. Dagger won't include `@Generated` when it detects the type is absent from the classpath.

File: sample/src/main/java/com/example/sqlbrite/todo/TodoComponent.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.example.sqlbrite.todo;
```
Comment (JakeWharton) [inline sample/src/main/java/com/example/sqlbrite/todo/TodoComponent.java]: Needs copyright header

File: sample/src/main/java/com/example/sqlbrite/todo/db/DbModule.java [hunk 5]
```diff
@@ -25,7 +25,7 @@
 import rx.schedulers.Schedulers;
 import timber.log.Timber;
 
-@Module(complete = false, library = true)
+@Module()
```
Comment (JakeWharton) [inline sample/src/main/java/com/example/sqlbrite/todo/db/DbModule.java]: remove parenthesis

**Comment by michaldrabik** [comment]
Thanks, updated with changes.

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**